src/db.rs

```diff
@@ -1,45 +1,71 @@
 use std::{
     fs::{remove_file, File},
     io::{Read, Write},
-    path::PathBuf,
-    str::FromStr,
+    path::{Path, PathBuf},
 };
 
-use anyhow::{Error, Result};
-use serde::{Deserialize, Serialize};
+use anyhow::Result;
+use serde::{de::DeserializeOwned, Serialize};
 
-#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
-pub struct FileWriter {
-    path: PathBuf,
+pub trait Storage<T> {
```

Comment:
As mentioned before, it's probably better to move the file/storage logic to the caller (either handle all the file reading by flutter, or move this `db.rs` file to `rust_lib_donationwallet`)

Reply:
Yeah I was leaving it here for consumer that may find it useful but you may be right and we'd rather be consistent and totally remove it.